### PR TITLE
feat: implement exchanges

### PR DESCRIPTION
**Pull request by dated**
## Summary

https://app.clickup.com/t/rp61gu

## Checklist

-   [ ] My changes look good in both light AND dark mode
-   [ ] The change is not hardcoded to a single network, but has multi-asset in mind
-   [ ] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: src/domains/exchange/components/ExchangeForm/ExchangeForm.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	//   from: "btc",
+	//   to: "ark",
+	//   amountFrom: 1,
+	//   amountTo: 1,
+	//   externalId: ""
+	// }
+
+	const [activeTab, setActiveTab] = useState(1);
+
+	const form = useForm<any>({
```
Comment (alesmit) [inline src/domains/exchange/components/ExchangeForm/ExchangeForm.tsx:55]: @dated any chance we can add an interface here to describe the form model?

File: src/domains/exchange/hooks/use-exchange-providers.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+					newExchange[camelCase(key) as string] = value;
+				}
+
+				return newExchange;
+			});
+
+		let exchanges: Exchange[];
+
+		try {
+			const body = (await httpClient.get(exchangeHost)).json();
```
Comment (alesmit) [inline src/domains/exchange/hooks/use-exchange-providers.ts:25]: If I'm not mistaken, in `fetch` we need to await also `response.json()`. Isn't the same here?

File: src/domains/exchange/components/ExchangeForm/StatusStep.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+	</div>
+);
+
+const ExchangeStatus = ({ status }: { status: OrderStatus }) => {
+	const { t } = useTranslation();
+
+	if (status === OrderStatus.Failed) {
+		return (
+			<Alert className="my-6" variant="danger">
+				Failed
```
Comment (alesmit) [inline src/domains/exchange/components/ExchangeForm/StatusStep.tsx:81]: Missing i18n?

File: src/domains/exchange/hooks/use-exchange-providers.ts (hunk 2 above)
Comment (dated) [reply src/domains/exchange/hooks/use-exchange-providers.ts:25]: this `json` just parses the body of the response.

File: src/domains/exchange/components/ExchangeForm/ExchangeForm.tsx (hunk 1 above)
Comment (dated) [reply src/domains/exchange/components/ExchangeForm/ExchangeForm.tsx:55]: will add that 👍🏻

File: src/domains/exchange/components/ExchangeForm/StatusStep.tsx (hunk 3 above)
Comment (dated) [reply src/domains/exchange/components/ExchangeForm/StatusStep.tsx:81]: :yesyes: error handling still needs to be improved, including these alerts.

File: src/domains/exchange/components/ExchangeForm/ExchangeForm.tsx (hunk 1 above)
Comment (alesmit) [reply src/domains/exchange/components/ExchangeForm/ExchangeForm.tsx:55]: thanks!

**Comment by samharperpittam** [comment]
https://user-images.githubusercontent.com/44671015/132296853-a70fe7f2-cac4-4ca4-9cd8-e59cc772ac74.mov

:sadup:

**Comment by faustbrian** [comment]
Task linked: [CU-rp61gu [Exchanges] add exchange service and form](https://app.clickup.com/t/rp61gu)

**Review by faustbrian — APPROVED** [review]

**Pull request merged**